src/systems/bit-media-frames.js

```diff
@@ -133,89 +157,152 @@ const snapToFrame = (() => {
+    const previewMaterial = new THREE.MeshBasicMaterial();
+    previewMaterial.side = THREE.DoubleSide;
+    previewMaterial.transparent = true;
+    previewMaterial.opacity = 0.5;
+
+    const geometry = createPlaneBufferGeometry(1, 1, 1, 1, TEXTURES_FLIP_Y);
+    previewMesh = new THREE.Mesh(geometry, previewMaterial);
+    previewMesh.material.map = srcMesh.material.map;
+    previewMesh.material.needsUpdate = true;
+    // Preview mesh UVs are set to accommodate textureLoader default, but video textures don't match this
+    previewMesh.scale.y *= TEXTURES_FLIP_Y !== previewMesh.material.map.flipY ? -ratio : ratio;
+  } else {
+    previewMesh = cloneObject3D(srcMesh, false);
+    previewMesh.traverse(node => {
+      updateMaterials(node, function (srcMat) {
+        const mat = srcMat.clone();
+        mat.transparent = true;
+        mat.opacity = 0.5;
+        mat.format = THREE.RGBAFormat;
+        mat.blending = THREE.NormalBlending;
+        node.material = mat;
+        return mat;
+      });
     });
-  });
 
-  // TODO play animations on previews
+    if (hasComponent(world, AEntity, capturable)) {
+      const loopAnimation = el.components["loop-animation"];
+      if (loopAnimation && loopAnimation.isPlaying) {
+        const originalAnimation = loopAnimation.currentActions[loopAnimation.data.activeClipIndex];
+        const animation = previewMesh.animations[loopAnimation.data.activeClipIndex];
+        const mixer = new THREE.AnimationMixer(previewMesh);
+        const action = mixer.clipAction(animation);
+        action.syncWith(originalAnimation);
+        action.setLoop(THREE.LoopRepeat, Infinity).play();
+        frame2mixer.set(frame, mixer);
+      }
+    }
+  }
 
   // TODO HACK We add this mesh to a group whose position is centered
   //      so that putting this in the middle of a media frame is easy,
   //      but we should just do this math when putting an object into a frame
   //      and not assume an object's root is in the center of its geometry.
-  meshClone.position.setScalar(0);
-  meshClone.quaternion.identity();
-  meshClone.matrixNeedsUpdate = true;
-  const aabb = new THREE.Box3().setFromObject(meshClone);
-  aabb.getCenter(meshClone.position).multiplyScalar(-1);
-  meshClone.matrixNeedsUpdate = true;
+  previewMesh.position.setScalar(0);
+  previewMesh.quaternion.identity();
+  previewMesh.matrixNeedsUpdate = true;
+  const aabb = new THREE.Box3().setFromObject(previewMesh);
+  aabb.getCenter(previewMesh.position).multiplyScalar(-1);
+  previewMesh.matrixNeedsUpdate = true;
 
   const cloneObj = new THREE.Group();
-  cloneObj.el = el; // We rely on media-loader component for bounds
-  cloneObj.add(meshClone);
-  AFRAME.scenes[0].object3D.add(cloneObj);
+  cloneObj.add(previewMesh);
+  APP.world.scene.add(cloneObj);
 
-  return addObject3DComponent(world, addEntity(world), cloneObj);
+  hasComponent(world, AEntity, capturable) && (cloneObj.el = el); // We rely on media-loader component for bounds
```

Comment:
I generally use it for simple conditions like:
```
A & B;
```
to avoid braces clutter. I don't have a strong opinion about it, if people find that makes the code hard to read I'm happy to stop using it.